internal/entities/properties/service/options.go

```diff
@@ -0,0 +1,113 @@
+//
+// Copyright 2024 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package service
+
+import "github.com/stacklok/minder/internal/db"
+
+// CallOptions is a struct that contains the options for a service call
+// Since most calls will need to interact with the database, the ExtendQuerier is included
+// to ensure we can pass in a transaction if needed.
+type CallOptions struct {
+	storeOrTransaction db.ExtendQuerier
+}
+
+// CallBuilder is a function that returns a new CallOptions struct
+func CallBuilder() *CallOptions {
+	return &CallOptions{}
+}
+
+// WithStoreOrTransaction is a function that sets the StoreOrTransaction field in the CallOptions struct
+func (psco *CallOptions) WithStoreOrTransaction(storeOrTransaction db.ExtendQuerier) *CallOptions {
+	if psco == nil {
+		return nil
+	}
+	psco.storeOrTransaction = storeOrTransaction
+	return psco
+}
+
+func (psco *CallOptions) getStoreOrTransaction() db.ExtendQuerier {
+	if psco == nil {
+		return nil
+	}
+	return psco.storeOrTransaction
+}
+
+// ReadOptions is a struct that contains the options for a read service call
+// This extends the PropertiesServiceCallOptions struct and adds a TolerateStaleData field.
+// This field is used to determine if the service call can return stale data or not.
+// This is useful for read calls that can tolerate stale data.
+type ReadOptions struct {
+	*CallOptions
```

Comment:
The callers read much nicer now, the only question I have is if you considered embedding CallOptions directly and not through a pointer. It seems like it would simplify the methods like WithStoreOrTransaction that now have to initialize CallOptions and then just forward, with an embedded struct you could forward directly.

Reply:
I can give it a try!